tests/unit/clustering/kmeans_unittest.cc

```diff
@@ -162,3 +165,111 @@ TEST(KMeans, minibatch_training_test)
 	SG_UNREF(features);
 }
 
+TEST(KMeans, fixed_centers)
+{
+	/*create a rectangle with four points as (0,0) (0,10) (20,0) (20,10)*/
+	SGMatrix<float64_t> rect(2, 4);
+	rect(0,0)=0;
+	rect(1,0)=0;
+	rect(0,1)=0;
+	rect(1,1)=10;
+	rect(0,2)=20;
+	rect(1,2)=0;
+	rect(0,3)=20;
+	rect(1,3)=10;
+
+	/*choose local minima points (but not exact means) (0,4) (20,4) as initial centers */
+	SGMatrix<float64_t> initial_centers(2,2);
+	initial_centers(0,0)=0;
+	initial_centers(1,0)=4;
+	initial_centers(0,1)=20;
+	initial_centers(1,1)=4;
+
+	CDenseFeatures<float64_t>* features=new CDenseFeatures<float64_t>(rect);
+	SG_REF(features);	
+	CEuclideanDistance* distance=new CEuclideanDistance(features, features);
+	CKMeans* clustering=new CKMeans(2, distance,initial_centers);
+	clustering->set_fixed_centers(true);
+
+	clustering->train(features);
+	CDenseFeatures<float64_t>* learnt_centers=CDenseFeatures<float64_t>::obtain_from_generic(distance->get_lhs());
```

Comment:
minor: in c++ you can cast.
This might return a NULL pointer if you do it this way. I guess our static code analysis might complain here.
So better put in an ASSERT_NE